lib/credentials/shared_ini_file_credentials.js

```diff
@@ -187,25 +200,33 @@ AWS.SharedIniFileCredentials = AWS.util.inherit(AWS.Credentials, {
   * @param profile [map] credentials profile
   * @throws SharedIniFileCredentialsProviderFailure
   */
-  loadViaCredentialProcess: function loadCredentialProcess(profile, callback) {
-    try {
-      var output = proc.execSync(profile['credential_process']);
-      var credData = JSON.parse(output);
-      if (parseInt(credData.Version) !== 1) {
-        throw AWS.util.error(
-          Error('credential_process does not return Version == 1 for profile ' + this.profile),
-          { code: 'SharedIniFileCredentialsProviderFailure'}
+ loadViaCredentialProcess: function loadViaCredentialProcess(profile, callback) {
+  proc.exec(profile['credential_process'], function(err,stdOut, stdErr) {
+    if (err) {
+      callback(err, null);
+    } else if (stdErr) {
+      callback(stdErr, null);
```

Comment:
stderr is often used for ancillary information (like a deprecation warning) even when the command succeeds, so I would advise against treating a truthy value as a failure.

FWIW, the [botocore implementation] ignores stderr.

If this is retained, I suggest passing `AWS.util.error(new Error(stdErr), {code: 'SharedIniCredentialsProviderFailue'})`, so that the callback receives a normalized `Error`, not a `string`.

[botocore implementation]: https://github.com/boto/botocore/blob/8d3ea0e61473fba43774eb3c74e1b22995ee7370/botocore/credentials.py#L804-L833

Reply:
Followed the botocore implementation and ignored stderr. thanks!